src/chat-client/index.ts

```diff
@@ -661,29 +658,102 @@ class Chat extends ClientBase {
+    const args = ['chat', 'api-listen']
+    if (this.homeDir) {
+      args.unshift('--home', this.homeDir)
+    }
+    if (!options || (options && options.hideExploding !== false)) {
+      args.push('--hide-exploding')
+    }
+    if (options && options.showLocal === true) {
+      args.push('--local')
+    }
+    if (channel) {
+      args.push('--filter-channel', JSON.stringify(formatAPIObjectInput(channel, 'chat')))
+    }
+    return args
+  }
+
+  /**
+   * Spawns the chat listen process and handles the calling of onMessage, onError, and filtering for a specific channel.
+   * @memberof Chat
+   * @ignore
+   * @param onMessage - A callback that is triggered on every message your bot receives.
+   * @param onError - A callback that is triggered on any error that occurs while the method is executing.
+   * @param channel - The chat channel to watch.
+   * @param options - Options for the listen method.
+   * @example
+   * this._chatListenMessage(onMessage, onError)
+   */
+  private _chatListenMessage(onMessage: OnMessage, onError?: OnError, channel?: chat1.ChatChannel, options?: ListenOptions): void {
+    const args = this._getChatListenArgs(channel, options)
     const onLine = (line: string): void => {
       this._adminDebugLogger.info(`stdout from listener: ${line}`)
       try {
-        const messageObject: chat1.MsgNotification = formatAPIObjectOutput(JSON.parse(line))
+        const messageObject = formatAPIObjectOutput(JSON.parse(line))
         if (messageObject.hasOwnProperty('error')) {
           throw new Error(messageObject.error)
-        } else if (
+        }
+        if (messageObject.type !== 'chat' || !messageObject.msg) {
+          return
+        }
+        const msgNotification: chat1.MsgNotification = messageObject
+        if (
           // fire onMessage if it was from a different sender or at least a different device
           // from this sender. Bots can filter out their own messages from other devices.
           (options && options.showLocal) ||
           (this.username &&
             this.devicename &&
-            (messageObject.msg.sender.username !== this.username.toLowerCase() || messageObject.msg.sender.deviceName !== this.devicename))
+            (msgNotification.msg.sender.username !== this.username.toLowerCase() ||
+              msgNotification.msg.sender.deviceName !== this.devicename))
         ) {
-          onMessage(messageObject.msg)
+          onMessage(msgNotification.msg)
         }
       } catch (error) {
         if (onError) {
           onError(error)
         }
       }
     }
-    lineReaderStdout.on('line', onLine)
+    this._adminDebugLogger.info(`spawningChatListenChild on channel=${JSON.stringify(channel || 'ALL')}`)
+    this._spawnChatListenChild(args, onLine)
+  }
+
+  /**
+   * Spawns the chat listen process for new channels and handles the calling of onConv, and onError.
```

Comment:
This listener will also receive new message notifications, why not just have a single listener that can take `onMessage` and `onConv` callbacks?

Reply:
The JS lib relies on the Go side to filter on channels. So we already have multiple api-listen calls happening if you watch multiple channels, or watch all messages and a particular channel at the same time. I guess we could use the same one for watching all messages and new conversations, and keep the channel watching the same way as before, but it didn’t seem worth it to me if it’s not the same form for everything.

Reply:
Gotcha makes sense!